tests/metaworld/envs/mujoco/sawyer_xyz/test_scripted_policies.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+
+from metaworld.envs.mujoco.env_dict import (
+    EASY_MODE_CLS_DICT, MEDIUM_MODE_CLS_DICT, HARD_MODE_CLS_DICT)
+from metaworld.policies import *
+from tests.metaworld.envs.mujoco.sawyer_xyz.utils import check_success
+
+
+test_data = [
+    # name,      policy,      success rate,   environment kwargs
+    ['door-v1', SawyerDoorOpenV1Policy(), 0.99, {}],
+    ['door-close-v1', SawyerDoorCloseV1Policy(), 0.99, {}],
+    ['drawer-open-v1', SawyerDrawerOpenV1Policy(), 0.99, {}],
+    ['drawer-close-v1', SawyerDrawerCloseV1Policy(), 0.99, {}]
+]
+
+ALL_ENVIRONMENTS = EASY_MODE_CLS_DICT
+ALL_ENVIRONMENTS.update(MEDIUM_MODE_CLS_DICT['train'])
+ALL_ENVIRONMENTS.update(MEDIUM_MODE_CLS_DICT['test'])
+ALL_ENVIRONMENTS.update(HARD_MODE_CLS_DICT['train'])
+ALL_ENVIRONMENTS.update(HARD_MODE_CLS_DICT['test'])
+
+for row in test_data:
+    row[-1] = ALL_ENVIRONMENTS[row[0]](random_init=True, **row[-1])
+
+
+@pytest.mark.parametrize('label,policy,expected_success_rate,env', test_data)
+def test_scripted_policy(env, policy, label, expected_success_rate, iters=1000):
+    """Tests whether a given policy solves an environment in a stateless manner
+
+    Args:
+        env (metaworld.envs.MujocoEnv): Environment to test
+        policy (metaworld.policies.policy.Policy): Policy that's supposed to succeed in env
+        label (string): A string to use when printing out errors
+        expected_success_rate (float): Decimal value indicating % of runs that must be successful
+        iters (int): How many times the policy should be tested
+
+    """
+    for name, value in vars(policy).items():
+        assert name == '_p', label + ' policy has potential state variable: ' + name
+
+    successes = 0
+    for i in range(iters):
+        successes += float(check_success(env, policy)[0])
+        if successes >= expected_success_rate * iters:
+            return
```

Comment:
tests should not return to avoid failure. they should always terminate or pass based on assertion conditions.